### PR TITLE
Recover hormud data

### PR DESCRIPTION
**Pull request by as2388**
Review #79 first.

Not sure if I should merge this given that the project finished without incorporating the recovered Hormud data into the analysis file used in the report, but this PR at least shows how data recovery can be done.

### CODE REVIEW

**Comment by lukechurch** [comment]
@as2388 Maybe we could tag the release that was used to produce the final analysis file?

**Comment by lukechurch** [comment]
@as2388 Please could you address the conflicts and request a review again?

**Review by lukechurch — CHANGES_REQUESTED** [review]
conflicts

**Comment by as2388** [comment]
@lukechurch conflicts resolved

File: run_scripts/2_fetch_data.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ done
 
 if [[ $# -ne 4 ]]; then
     echo "Usage: ./2_fetch_raw_data.sh [--profile-cpu <cpu-profile-output-path>] <user> <google-cloud-credentials-file-path> <pipeline-configuration-file-path> <data-root>"
-    echo "Fetches all the raw data from Rapid Pro and converts to TracedData"
+    echo "Fetches all the raw data from Rapid Pro and from the Shaqadoon recovery CSVS, and converts to TracedData"
```
Comment (lukechurch) [inline run_scripts/2_fetch_data.sh]: nit: CSVs

File: generate_outputs.py [hunk 2]
```diff
@@ -102,6 +102,20 @@
...
         messages_datasets.append(messages)
 
+    recovery_datasets = []
+    if pipeline_configuration.recovery_csv_urls is None:
+        log.debug("Not loading any recovery datasets (because the pipeline configuration json does not contain the key "
+                  "'RecoveryCSVURLs')")
+    else:
+        log.info("Loading recovery datasets:")
+        for i, recovery_csv_url in enumerate(pipeline_configuration.recovery_csv_urls):
+            raw_recovery_path = f"{raw_data_dir}/{recovery_csv_url.split('/')[-1].split('.')[0]}.json"
```
Comment (lukechurch) [inline generate_outputs.py:112]: this feels a bit unpleasant, but it also doesn't feel worth a different structural solution.

**Review by lukechurch — APPROVED** [review]
Thanks @as2388

**Pull request merged**